table/merging_iterator.cc

```diff
@@ -1177,7 +1092,7 @@ void MergingIterator::SwitchToForward() {
       if (child.iter.status() == Status::TryAgain()) {
         continue;
       }
-      if (child.iter.Valid() && comparator_->Equal(target, child.key())) {
+      if (child.iter.Valid() && comparator_->Equal(target, child.iter.key())) {
```

Comment:
Are we using `child.iter.key()` to avoid `HeapItem::key()` because its implementation is specific to `CompactionMergingIterator`? It's sort of confusing

Reply:
Yes that was the confusing intention. MergingIterator cannot use key() for both point keys and range tombstones keys since they are of different types. I've update the code for MergingIterator and CompactionMergingIterator to each have its own `HeapItem` structure.